### PR TITLE
Prepare to introduce websockets for exec and portforward

### PR DESCRIPTION
**Pull request by smarterclayton**
Refactor the code in remotecommand to better represent the structure of
what is common between portforward and exec.

Ref #48633

### CODE REVIEW

**Comment by smarterclayton** [comment]
@ncdc this is the great refactor, long in the making!  I'm just going to do some small cleanup and code movement before I add the abstraction for websocket.

File: staging/src/k8s.io/client-go/transport/spdy/spdy.go [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+	}
+	upgradeRoundTripper := spdy.NewRoundTripper(tlsConfig, true)
+	wrapper, err := restclient.HTTPWrappersForConfig(config, upgradeRoundTripper)
+	if err != nil {
+		return nil, nil, err
+	}
+	return wrapper, upgradeRoundTripper, nil
+}
+
+// spdyDialer implements the httpstream.Dialer interface.
```
Comment (ncdc) [inline staging/src/k8s.io/client-go/transport/spdy/spdy.go]: s/spdyDialer/dialer/

**Review by ncdc — CHANGES_REQUESTED** [review]
LGTM overall

**Comment by ncdc** [comment]
Need to fix bazel

**Comment by smarterclayton** [comment]
Updated (except for the kubefed issue which is not related)

**Comment by smarterclayton** [comment]
Applying label per comments and green (andy correct me if you wanted more)

**Review by ncdc — APPROVED** [review]

**Comment by ncdc** [comment]
Good to go, thanks

**Pull request merged**